app/routes/session.py

```diff
@@ -151,76 +129,17 @@ def login() -> Response:
 
     cookie_session["language_code"] = metadata.language_code
 
-    if (dataset_id := metadata["sds_dataset_id"]) and ru_ref:
+    # Type ignore: survey_id and either ru_ref or qid are required for schemas that use supplementary data
+    if dataset_id := metadata["sds_dataset_id"]:
         get_supplementary_data(
-            supplementary_data_url=SUPPLEMENTARY_DATA_URL,
             dataset_id=dataset_id,
-            ru_ref=ru_ref,
+            unit_id=metadata["ru_ref"] or metadata["qid"],  # type: ignore
```

Comment:
do we have validation that its always exactly one of these that is present? I guess we do in validator but I think it might be good here as well?